File: esports-backend/src/main/java/com/esports/controller/AdminController.java
package com.esports.controller;

import com.esports.dto.*;
import com.esports.service.AdminService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;
import jakarta.validation.Valid;

import java.util.List;
import java.util.NoSuchElementException;

@RestController
@RequestMapping("/api/admin")
@CrossOrigin(origins = "http://localhost:5173")
@PreAuthorize("hasRole('ADMIN')")
public class AdminController {

    @Autowired
    private AdminService adminService;

    @GetMapping("/stats")
    public ResponseEntity<AdminStatsResponseDTO> getAdminStats() {
        try {
            AdminStatsResponseDTO stats = adminService.getAdminStats();
            return ResponseEntity.ok(stats);
        } catch (Exception e) {
            return ResponseEntity.status(500).body(null);
        }
    }

    @GetMapping("/kyc/pending")
    public ResponseEntity<List<KycDocumentResponseDTO>> getPendingKYCDocuments() {
        try {
            List<KycDocumentResponseDTO> documents = adminService.getPendingKycDocuments();
            return ResponseEntity.ok(documents);
        } catch (Exception e) {
            return ResponseEntity.status(500).body(null);
        }
    }

    @PostMapping("/kyc/review")
    public ResponseEntity<ApiResponse<String>> reviewKYCDocument(@RequestBody KycReviewRequestDTO request) {
        try {
            adminService.reviewKycDocument(request);
            return ResponseEntity.ok(ApiResponse.success("Document " + request.getAction() + "d successfully"));
        } catch (NumberFormatException e) {
            return ResponseEntity.status(400).body(ApiResponse.error("Invalid request format"));
        } catch (NoSuchElementException e) {
            return ResponseEntity.status(404).body(ApiResponse.error("Verification not found"));
        } catch (IllegalArgumentException e) {
            return ResponseEntity.status(400).body(ApiResponse.error(e.getMessage()));
        } catch (Exception e) {
            return ResponseEntity.status(500).body(ApiResponse.error("Failed to review document"));
        }
    }

    @GetMapping("/users")
    public ResponseEntity<List<AdminUserResponseDTO>> getAllUsers() {
        try {
            List<AdminUserResponseDTO> users = adminService.getAllUsers();
            return ResponseEntity.ok(users);
        } catch (Exception e) {
            return ResponseEntity.status(500).body(null);
        }
    }

    @PutMapping("/users/{userId}/deactivate")
    public ResponseEntity<ApiResponse<String>> deactivateUser(@PathVariable Long userId) {
        try {
            adminService.deactivateUser(userId);
            return ResponseEntity.ok(ApiResponse.success("User deactivated successfully"));
        } catch (NoSuchElementException e) {
            return ResponseEntity.status(404).body(ApiResponse.error("User not found"));
        } catch (IllegalArgumentException e) {
            return ResponseEntity.status(400).body(ApiResponse.error(e.getMessage()));
        } catch (Exception e) {
            return ResponseEntity.status(500).body(ApiResponse.error("Failed to deactivate user"));
        }
    }

    @PutMapping("/users/{userId}/activate")
    public ResponseEntity<ApiResponse<String>> activateUser(@PathVariable Long userId) {
        try {
            adminService.activateUser(userId);
            return ResponseEntity.ok(ApiResponse.success("User activated successfully"));
        } catch (NoSuchElementException e) {
            return ResponseEntity.status(404).body(ApiResponse.error("User not found"));
        } catch (Exception e) {
            return ResponseEntity.status(500).body(ApiResponse.error("Failed to activate user"));
        }
    }

    @PutMapping("/users/{userId}")
    public ResponseEntity<ApiResponse<String>> updateUser(@PathVariable Long userId, @Valid @RequestBody UserUpdateRequestDTO request) {
        try {
            adminService.updateUser(userId, request);
            return ResponseEntity.ok(ApiResponse.success("User updated successfully"));
        } catch (NoSuchElementException e) {
            return ResponseEntity.status(404).body(ApiResponse.error("User not found"));
        } catch (IllegalArgumentException e) {
            return ResponseEntity.status(400).body(ApiResponse.error(e.getMessage()));
        } catch (Exception e) {
            return ResponseEntity.status(500).body(ApiResponse.error("Failed to update user"));
        }
    }

    @PutMapping("/users/{userId}/kyc-status")
    public ResponseEntity<ApiResponse<String>> updateUserKycStatus(@PathVariable Long userId, @RequestBody KycStatusUpdateDTO request) {
        try {
            adminService.updateUserKycStatus(userId, request.getKycStatus());
            return ResponseEntity.ok(ApiResponse.success("KYC status updated successfully"));
        } catch (NoSuchElementException e) {
            return ResponseEntity.status(404).body(ApiResponse.error("User not found"));
        } catch (IllegalArgumentException e) {
            return ResponseEntity.status(400).body(ApiResponse.error(e.getMessage()));
        } catch (Exception e) {
            return ResponseEntity.status(500).body(ApiResponse.error("Failed to update KYC status"));
        }
    }

    @GetMapping("/users/{userId}/documents")
    public ResponseEntity<List<UserDocumentResponseDTO>> getUserDocuments(@PathVariable Long userId) {
        try {
            List<UserDocumentResponseDTO> documents = adminService.getUserDocuments(userId);
            return ResponseEntity.ok(documents);
        } catch (NoSuchElementException e) {
            return ResponseEntity.status(404).body(null);
        } catch (Exception e) {
            return ResponseEntity.status(500).body(null);
        }
    }
}

File: Upkart1/Controllers/AdminController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Upkart1.DTO;
using Upkart1.Services;
using Upkart1.Filters;

namespace Upkart1.Controllers
{
    [ApiController]
    [Route("api/admin")]
    [AdminOnly]
    public class AdminController : ControllerBase
    {
        private readonly IAdminService _adminService;

        public AdminController(IAdminService adminService)
        {
            _adminService = adminService;
        }

        [HttpGet("stats")]
        public IActionResult GetAdminStats()
        {
            try
            {
                var stats = _adminService.GetAdminStats();
                return Ok(stats);
            }
            catch (Exception)
            {
                return StatusCode(500, null);
            }
        }

        [HttpGet("kyc/pending")]
        public IActionResult GetPendingKYCDocuments()
        {
            try
            {
                var documents = _adminService.GetPendingKycDocuments();
                return Ok(documents);
            }
            catch (Exception)
            {
                return StatusCode(500, null);
            }
        }

        [HttpPost("kyc/review")]
        public IActionResult ReviewKYCDocument([FromBody] KycReviewRequestDTO request)
        {
            try
            {
                _adminService.ReviewKycDocument(request);
                return Ok(ApiResponse<string>.SuccessResponse($"Document {request.Action}d successfully"));
            }
            catch (FormatException)
            {
                return BadRequest(ApiResponse<string>.ErrorResponse("Invalid request format"));
            }
            catch (KeyNotFoundException)
            {
                return NotFound(ApiResponse<string>.ErrorResponse("Verification not found"));
            }
            catch (ArgumentException ex)
            {
                return BadRequest(ApiResponse<string>.ErrorResponse(ex.Message));
            }
            catch (Exception)
            {
                return StatusCode(500, ApiResponse<string>.ErrorResponse("Failed to review document"));
            }
        }

        [HttpGet("users")]
        public IActionResult GetAllUsers()
        {
            try
            {
                var users = _adminService.GetAllUsers();
                return Ok(users);
            }
            catch (Exception)
            {
                return StatusCode(500, null);
            }
        }

        [HttpPut("users/{userId}/deactivate")]
        public IActionResult DeactivateUser(long userId)
        {
            try
            {
                _adminService.DeactivateUser(userId);
                return Ok(ApiResponse<string>.SuccessResponse("User deactivated successfully"));
            }
            catch (KeyNotFoundException)
            {
                return NotFound(ApiResponse<string>.ErrorResponse("User not found"));
            }
            catch (ArgumentException ex)
            {
                return BadRequest(ApiResponse<string>.ErrorResponse(ex.Message));
            }
            catch (Exception)
            {
                return StatusCode(500, ApiResponse<string>.ErrorResponse("Failed to deactivate user"));
            }
        }

        [HttpPut("users/{userId}/activate")]
        public IActionResult ActivateUser(long userId)
        {
            try
            {
                _adminService.ActivateUser(userId);
                return Ok(ApiResponse<string>.SuccessResponse("User activated successfully"));
            }
            catch (KeyNotFoundException)
            {
                return NotFound(ApiResponse<string>.ErrorResponse("User not found"));
            }
            catch (Exception)
            {
                return StatusCode(500, ApiResponse<string>.ErrorResponse("Failed to activate user"));
            }
        }

        [HttpPut("users/{userId}")]
        public IActionResult UpdateUser(long userId, [FromBody] UserUpdateRequestDTO request)
        {
            try
            {
                _adminService.UpdateUser(userId, request);
                return Ok(ApiResponse<string>.SuccessResponse("User updated successfully"));
            }
            catch (KeyNotFoundException)
            {
                return NotFound(ApiResponse<string>.ErrorResponse("User not found"));
            }
            catch (ArgumentException ex)
            {
                return BadRequest(ApiResponse<string>.ErrorResponse(ex.Message));
            }
            catch (Exception)
            {
                return StatusCode(500, ApiResponse<string>.ErrorResponse("Failed to update user"));
            }
        }

        [HttpPut("users/{userId}/kyc-status")]
        public IActionResult UpdateUserKycStatus(long userId, [FromBody] KycStatusUpdateDTO request)
        {
            try
            {
                _adminService.UpdateUserKycStatus(userId, request.KycStatus);
                return Ok(ApiResponse<string>.SuccessResponse("KYC status updated successfully"));
            }
            catch (KeyNotFoundException)
            {
                return NotFound(ApiResponse<string>.ErrorResponse("User not found"));
            }
            catch (ArgumentException ex)
            {
                return BadRequest(ApiResponse<string>.ErrorResponse(ex.Message));
            }
            catch (Exception)
            {
                return StatusCode(500, ApiResponse<string>.ErrorResponse("Failed to update KYC status"));
            }
        }

        [HttpGet("users/{userId}/documents")]
        public IActionResult GetUserDocuments(long userId)
        {
            try
            {
                var documents = _adminService.GetUserDocuments(userId);
                return Ok(documents);
            }
            catch (KeyNotFoundException)
            {
                return NotFound(null);
            }
            catch (Exception)
            {
                return StatusCode(500, null);
            }
        }
    }
}
